src/Context.Tests/Extensions/MongoDatabaseExtensionsTests.cs

```diff
@@ -0,0 +1,144 @@
+
+        [Fact]
+        public void EnableProfiling_EnableDefaultProfileLevel_EnabledProfilingOnLevel2()
+        {
+            // Arrange
+
+            // Act
+            _mongoDatabase.EnableProfiling();
+
+            // Assert
+            var profileStatusCommand = new BsonDocument("profile", BsonValue.Create(null));
+            BsonDocument result = _mongoDatabase.RunCommand<BsonDocument>(profileStatusCommand);
+
+            Assert.Equal(
+                "{ \"was\" : 2, \"slowms\" : 100, \"sampleRate\" : 1.0, \"ok\" : 1.0 }",
+                result.ToString());
+        }
+
+        [Fact]
+        public void EnableProfiling_EnableAllProfileLevel_EnabledProfilingOnLevel2()
+        {
+            // Arrange
+
+            // Act
+            _mongoDatabase.EnableProfiling(ProfileLevel.All);
+
+            // Assert
+            var profileStatusCommand = new BsonDocument("profile", BsonValue.Create(null));
+            BsonDocument result = _mongoDatabase.RunCommand<BsonDocument>(profileStatusCommand);
+
+            Assert.Equal(
+                "{ \"was\" : 2, \"slowms\" : 100, \"sampleRate\" : 1.0, \"ok\" : 1.0 }",
+                result.ToString());
+        }
+
+        [Fact]
+        public void EnableProfiling_EnableSlowOperationsOnlyProfileLevel_EnabledProfilingOnLevel1()
+        {
+            // Arrange
+
+            // Act
+            _mongoDatabase.EnableProfiling(ProfileLevel.SlowOperationsOnly);
+
+            // Assert
+            var profileStatusCommand = new BsonDocument("profile", BsonValue.Create(null));
+            BsonDocument result = _mongoDatabase.RunCommand<BsonDocument>(profileStatusCommand);
+
+            Assert.Equal(
+                "{ \"was\" : 1, \"slowms\" : 100, \"sampleRate\" : 1.0, \"ok\" : 1.0 }",
+                result.ToString());
+        }
+
+        [Fact]
+        public void EnableProfiling_DisableProfileLevel_EnabledProfilingOnLevel1()
+        {
+            // Arrange
+
+            // Act
+            _mongoDatabase.EnableProfiling(ProfileLevel.Off);
+
+            // Assert
+            var profileStatusCommand = new BsonDocument("profile", BsonValue.Create(null));
+            BsonDocument result = _mongoDatabase.RunCommand<BsonDocument>(profileStatusCommand);
+
+            Assert.Equal(
+                "{ \"was\" : 0, \"slowms\" : 100, \"sampleRate\" : 1.0, \"ok\" : 1.0 }",
+                result.ToString());
+        }
+
+        #endregion
+
+        #region EnableProfiling Tests
+
+        [Fact]
+        public void GetProfilingStatus_GetDisabledProfileStatus_StatusDisabled()
+        {
+            // Arrange
+            var profileCommand = new BsonDocument("profile", 0);
+            _mongoDatabase.RunCommand<BsonDocument>(profileCommand);
+
+            // Act
+            ProfilingStatus profileStatus = _mongoDatabase.GetProfilingStatus();
+
+            // Assert
+            Assert.Equal(ProfileLevel.Off, profileStatus.Level);
```

Comment:
4 aserts ... Maybe usage of Snapshooter ;)

Reply:
Its an integration test, therefore allowed :) Yeah, I was thinking about snapshooter, but sometimes snapshots are overused....